ts4.1/index.d.ts

```diff
@@ -100,7 +100,7 @@ declare module 'i18next' {
 }
 
 type ObjectOrNever = TypeOptions['allowObjectInHTMLChildren'] extends true
-  ? Record<string, unknown>
```

Comment:
I think this would be cleaner:
```
type HTMLChild = ReactNode | ObjectOrNever;
declare module 'react' {
  interface HTMLAttributes<T> {
    children?: HTMLChild | Iterable<HTMLChild>;
  }
}
```

I've used `Iterable` because that's what react itself uses for `ReactNode`'s circular reference, and given the type a specific name so that `ObjectOrNever` still actually represents `ObjectOrNever`.

Reply:
True, updated MR. Thx for the suggestion!

Reply:
Interestingly enough `Iterable` does not work correctly when combined with `t` function, but `Array` works as expected 🤔 
Here's a code example:
```tsx
<div>
    {t('hello')}
    <span>Hi</span>
</div>
```
I get a compilation error on outer div

Reply:
Either disabling `allowObjectInHTML` or changing `Iterable` to `Array` fixes it